src/lando/api/legacy/transplants.py

```diff
@@ -866,10 +854,15 @@ def blocker_single_landing_repo(
     if not repo_phid:
         return "Landing path has no repository specified."
 
-    # Set the landing repo field on the `LandingAssessmentState`.
-    landing_repo = stack_state.landable_repos.get(repo_phid)
+    if repo_phid not in stack_state.landable_repos:
+        return "Landing repository is not supported by Lando."
 
-    stack_state.landing_assessment.landing_repo = landing_repo
+    # If the landing repo on the current revision is set as a legacy repo
+    # of another repo, then change the landing repo to match the target.
+    set_repo = stack_state.landable_repos[repo_phid]
+    stack_state.landing_assessment.landing_repo = (
+        set_repo if not set_repo.is_legacy else set_repo.new_target
+    )
```

Comment:
Not sure if this will have impact on functionality at this time, but it's a little odd that `stack_state.landing_assessment.landing_repo` will have different values in different checks. Maybe these should be set elsewhere to apply to all checks? Maybe there should be something like `stack_state.landing_repo_mapping` which could already have the correct mapping from a revision PHID to a target repo.

Reply:
I filed [bug 1951169](https://bugzilla.mozilla.org/show_bug.cgi?id=1951169) to track this.